src/ast/mod.rs

```diff
@@ -437,19 +437,22 @@ pub enum Expr {
     Cast {
         expr: Box<Expr>,
         data_type: DataType,
+        format: Option<Value>,
     },
     /// TRY_CAST an expression to a different data type e.g. `TRY_CAST(foo AS VARCHAR(123))`
     //  this differs from CAST in the choice of how to implement invalid conversions
     TryCast {
         expr: Box<Expr>,
         data_type: DataType,
+        format: Option<Value>,
```

Comment:
FWIW it appears bigquery supports a more sophisticated format clause than just a single value:
https://cloud.google.com/bigquery/docs/reference/standard-sql/format-elements#formatting_syntax

Perhaps we should create a `CastFormat` enum so if we want to extend the supported syntax the API change is less. Something like:

```suggestion
        format: Option<CastFormat>,
```

```rust
/// Options for `CAST` / `TRY_CAST` 
/// BigQuery: https://cloud.google.com/bigquery/docs/reference/standard-sql/format-elements#formatting_syntax
enum CastFormat {
   /// Constant string value
  String(Value)
 }
```

Reply:
(I don't feel strongly about this -- we can merge it as is as well)

Reply:
I think this is really good idea, I added CastFormat and covered `AT TIME ZONE` support too.